http/src/main/scala/org/broadinstitute/dsde/workbench/leonardo/monitor/LeoPubsubMessageSubscriber.scala

```diff
@@ -415,6 +422,51 @@ class LeoPubsubMessageSubscriber[F[_]: Timer: ContextShift](
       _ <- gceRuntimeMonitor.process(runtime.id).compile.drain
     } yield ()
 
+  private[monitor] def handleCreateDiskMessage(msg: CreateDiskMessage, now: Instant)(
+    implicit traceId: ApplicativeAsk[F, AppContext]
+  ): F[Unit] =
+    for {
+      _ <- googleDiskService.createDisk(
+        msg.googleProject,
+        msg.zone,
+        Disk
+          .newBuilder()
+          .setId(msg.diskId.toString)
+          .setCreationTimestamp(now.toString)
+          .setName(msg.name.toString)
+          .setSizeGb(msg.size.gb.toString)
+          .setZone(msg.zone.toString)
+          .build()
+      )
+    } yield ()
+
+  private[monitor] def handleDeleteDiskMessage(msg: DeleteDiskMessage, now: Instant)(
+    implicit traceId: ApplicativeAsk[F, AppContext]
+  ): F[Unit] =
+    for {
+      diskOpt <- persistentDiskQuery.getById(msg.diskId).transaction
+      disk <- diskOpt.fold(
+        F.raiseError[PersistentDisk](PubsubHandleMessageError.DiskNotFound(msg.diskId, msg))
+      )(F.pure)
+      // TODO: do I need to check disk status before deleting?
+      _ <- googleDiskService.deleteDisk(disk.googleProject, disk.zone, disk.name)
+    } yield ()
+
+  private[monitor] def handleUpdateDiskMessage(msg: UpdateDiskMessage, now: Instant)(
+    implicit traceId: ApplicativeAsk[F, AppContext]
+  ): F[Unit] =
+    for {
+      diskOpt <- persistentDiskQuery.getById(msg.diskId).transaction
+      disk <- diskOpt.fold(
+        F.raiseError[PersistentDisk](PubsubHandleMessageError.DiskNotFound(msg.diskId, msg))
+      )(F.pure)
+      _ <- if (msg.newSize.isDefined) {
```

Comment:
Maybe 
```
_ <- msg.newSize.traverse_ { newSize =>
  googleDiskService.resizeDisk(disk.googleProject, disk.zone, disk.name, newSize.gb)
}
```

Reply:
this is no longer an optional field so no need to check anymore if it exists :)